qiskit_experiments/analysis/plotting.py

```diff
@@ -81,18 +88,24 @@ def plot_curve_fit(
     ax.plot(xs, ys_fit, **plot_opts)
 
     # Plot standard error interval
-    if confidence_interval and fit_errors is not None:
+    if monotonicity and fit_errors is not None:
         if param_keys:
             params_upper = {}
             params_lower = {}
             for key, param, error in zip(param_keys, fit_params, fit_errors):
-                params_upper[key] = param + error
-                params_lower[key] = param - error
+                params_upper[key] = param - error * (-1) ** monotonicity[key]
+                params_lower[key] = param + error * (-1) ** monotonicity[key]
             ys_upper = func(xs, **params_upper)
             ys_lower = func(xs, **params_lower)
         else:
-            params_upper = [param + error for param, error in zip(fit_params, fit_errors)]
-            params_lower = [param - error for param, error in zip(fit_params, fit_errors)]
+            params_upper = [
```

Comment:
It looks like the same computation is repeated twice, for two different presentation of the input. Maybe it's better that in one presentation we simply restructure the input to fit the second presentation and run the computation once; otherwise every computation-related bugfix will need to be done twice, and we might miss the second one.

Reply:
I thought about it, however didn't find a way to perform the translation between the representations